library/bits.lisp

```diff
@@ -36,7 +39,25 @@
     (not   "The bitwise logical `not` of two integers"
            (:int -> :int))
     (shift "The arithmetic left-shift of an integer by an integer number of bits"
-           (Integer -> :int -> :int))))
+           (Integer -> :int -> :int)))
+
+  (declare dpb (Bits :a => :a -> UFix -> UFix -> :a -> :a))
+  (define (dpb newbyte size position bitstring)
+    "Deposits a byte `newbyte` of size `size` into a bitstring `bitstring` at a position `position`."
+    (lisp :a (newbyte bitstring size position)
+      (cl:dpb newbyte (cl:byte size position) bitstring)))
+
+  (declare ldb (Bits :a => UFix -> UFix -> :a -> :a))
+  (define (ldb size position bitstring)
+    "Deposits a byte of size `size` into a bitstring at a position `position`."
+    (lisp :a (bitstring size position)
+      (cl:ldb (cl:byte size position) bitstring)))
+
+  (declare bits->string (Bits :a => UFix -> :a -> String))
+  (define (bits->string width x)
```

Comment:
since x is a type class constrained type, it's not guaranteed to be a compatible Lisp value with FORMAT. you need to convert it to an integer for this to work, or extract the bits one by one 

i suggest either fixing this or removing it

Reply:
Sounds good, I removed it. 

It would fit better in a potential future Coalton formatting library anyways.